### PR TITLE
Refact

### PR DESCRIPTION
**Pull request by xylographe**
[2] Loop with arbitrary number of iterations replaced by loop with end-condition.

### CODE REVIEW

File: src/Forms/MultipleReplace.cs [hunk 1]
```diff
@@ -193,11 +193,7 @@ private void GeneratePreview()
                             int index = newText.IndexOf(findWhat, StringComparison.OrdinalIgnoreCase);
                             while (index >= 0)
                             {
-                                if (index < newText.Length)
-                                    newText = newText.Substring(0, index) + replaceWith + newText.Substring(index + findWhat.Length);
-                                else
-                                    newText = newText.Substring(0, index) + replaceWith;
-
+                                newText = newText.Substring(0, index) + replaceWith + newText.Remove(0, index + findWhat.Length);
```
Comment (xylographe) [inline src/Forms/MultipleReplace.cs]: I think, the intention was `if (index + findWhat.Length < newText.Length)`. But, this shouldn't be necessary, as the potential maximum value equals `newText.Length`.

File: src/Logic/StripableText.cs [hunk 2]
```diff
@@ -60,20 +63,24 @@ public StripableText(string text, string stripStartCharacters, string stripEndCh
...
+                    beginLength = text.Length;
+
                     while (text.Length > 0 && stripEndCharacters.Contains(text[text.Length - 1]))
                     {
                         Post = text[text.Length - 1] + Post;
                         text = text.Substring(0, text.Length - 1);
                     }
 
-                    if (text.EndsWith('>'))
+                    if (text.Length >= 4 && text[text.Length - 1] == '>')
```
Comment (xylographe) [inline src/Logic/StripableText.cs]: Not faster than EndsWith(). But, requiring `Length >= 4` eliminates text comparisons that are bound to fail.

Comment (xylographe) [inline src/Logic/StripableText.cs]: Actually, this was a bad idea! *Reverted*.

File: src/Forms/MultipleReplace.cs (hunk 1 above)
Comment (niksedk) [inline src/Forms/MultipleReplace.cs]: :)

**Pull request merged**